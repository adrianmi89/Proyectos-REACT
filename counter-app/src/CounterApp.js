import React,{useState} from "react";
import PropTypes from "prop-types";

//Las funciones de eventos por defecto reciben un argumento
//Se puede poner la funciÃ³n entera en el atributo u poner solo la referencia a la misma
const CounterApp = ({value}) => {
    
    const [ counter, setCounter ] = useState(value);//El useState siempre va tener un array

    const handleAdd = function(e) {

        setCounter( counter+1 );
    }
    const handleDelete = function(e) {

        setCounter( counter-1 );
    }
    const handleReset = function(e) {

        setCounter( counter=>0 );
    }
    return (
        <>
        <h1>CounterApp</h1>
        <h2> {counter} </h2>

        <button onClick={ handleAdd }>+</button>
        <button onClick={ handleDelete }>-</button>
        <button onClick={ handleReset }>Borrar</button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number
}

export default CounterApp;
