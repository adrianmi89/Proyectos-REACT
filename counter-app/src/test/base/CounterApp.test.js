//Tarea
/*
 1. Se debe de mostrar el programa correctamente haciendo match con un snapshot y
    sus valores por defecto.
 2. Se debe de mostrar el valor por defecto de 100. Usar el wrapper.find, tomando
    el elemento HTML donde se muestra el valor del contador. 
*/
import React from "react";
import "@testing-library/jest-dom";
import {shallow} from "enzyme";//El shallow muy parecido al render pero con algunas ventajas
import CounterApp from "../../CounterApp";

describe("Pruebas con el CounterApp",() => {
    /*
    test("Debe de mostrar el mensaje 'Hola, me llamo Link'",(done) => {

       const saludo = "me llamo link.";
       const {getByText} = render(<PrimeraApp persona={saludo} />);//Función del objeto wrapper

       expect (getByText(saludo)).toBeInTheDocument();

       done();
    })
    */
        const contador = 100;
        let wrapper = shallow(<CounterApp />);

        beforeEach(() => {
            
            wrapper = shallow(<CounterApp />);
        })
    test("Debe de mostrar <CounterApp /> correctamente haciendo un Snapshot y con el valor 100",() => {
        
        const wrapper = shallow(<CounterApp value={contador}/>);
        const nVisitas = wrapper.find("h2").text();

        console.log(nVisitas);

        expect(wrapper).toMatchSnapshot();
    })
    test("Debe de mostrar el valor por defecto del CounterApp",() => {
        
        const contador = 100;
        const wrapper = shallow(<CounterApp value={contador}/>);
        const nVisitas = wrapper.find("h2").text().trim();//Para quitarle los posibles espacios

        expect(nVisitas).toBe("100");
    })
    test("Debe de incrementar el contador a +1",() => {
        
        const wrapper = shallow(<CounterApp value = {contador}/>);

        const mas = wrapper.find("button").at(0).simulate("click");
        const nVisitas = wrapper.find("h2").text().trim();

        expect(nVisitas).toBe("101");
        console.log(mas.html());//Todas las propiedades en forma de función
        console.log(nVisitas);
    })
    test("Debe de decrementar el contador a -1",() => {
        
        const wrapper = shallow(<CounterApp value = {contador}/>);

        const menos = wrapper.find("button").at(1).simulate("click");
        const nVisitas = wrapper.find("h2").text().trim();

        expect(nVisitas).toBe("99");
        console.log(menos.html());//Todas las propiedades en forma de función
        console.log(nVisitas);
    })
    test("Debe de colocar el valor por defecto con botón reset",() => {
        
        const wrapper = shallow(<CounterApp value = {contador}/>);

        const reset = wrapper.find("button").at(2).simulate("click");
        const nVisitas = wrapper.find("h2").text().trim();

        expect(nVisitas).toBe("0");
        console.log(reset.html());//Todas las propiedades en forma de función
        console.log(nVisitas);
    })
})