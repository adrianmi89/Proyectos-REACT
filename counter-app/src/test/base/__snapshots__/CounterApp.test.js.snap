// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el CounterApp Debe de mostrar <CounterApp /> correctamente haciendo un Snapshot y con el valor 100 1`] = `ShallowWrapper {}`;

exports[`Pruebas con el CounterApp Debe de mostrar el valor por defecto del CounterApp 1`] = `ShallowWrapper {}`;
